client/state/themes/selectors.js

```diff
@@ -602,6 +602,23 @@ export function isPremiumThemeAvailable( state, themeId, siteId ) {
 }
 
 /**
+ * Whether a given theme is installed or can be installed on a Jetpack site.
+ *
+ * @param  {Object}  state   Global state tree
+ * @param  {String}  themeId Theme ID for which we check availability
+ * @param  {Number}  siteId  Site ID
+ * @return {Boolean}         True if siteId is a Jetpack site on which theme is installed or can be installed.
+ */
+export function isThemeAvailableOnJetpackSite( state, themeId, siteId ) {
```

Comment:
Also, I think you said on Slack that 'Activate' worked for you (showing non-JP sites). But the option is sharing this same criterion, so it can't be the culprit, can it?